File: golang/linked_list/partition_list.go
package main

func partition(head *ListNode, x int) *ListNode {
	leftDummy := &ListNode{}
	rightDummy := &ListNode{}

	left := leftDummy
	right := rightDummy

	current := head

	for current != nil {
		if current.Val < x {
			left.Next = current
			left = left.Next
		} else {
			right.Next = current
			right = right.Next
		}

		current = current.Next
	}

	right.Next = nil
	left.Next = rightDummy.Next

	return leftDummy.Next
}

File: python/linked_list/partition_list.py
from typing import Optional

class Solution:
    def partition(self, head: Optional[ListNode], x: int) -> Optional[ListNode]:

        left_dummy = ListNode(0)
        right_dummy = ListNode(0)


        left = left_dummy
        right = right_dummy


        current = head
        while current:
            if current.val < x:
                left.next = current
                left = left.next
            else:
                right.next = current
                right = right.next
            current = current.next


        right.next = None
        left.next = right_dummy.next

        return left_dummy.next
